src/history/hs_rec.c

```diff
@@ -189,28 +186,12 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
     if (ret == 0)
         WT_ERR_NOTFOUND_OK(cursor->next(cursor), true);
     else {
-        F_SET(cursor, WT_CURSTD_HS_READ_ALL);
```

Comment:
We added this flag to fix a massive performance regression when inserting records to history store.